VimeoNetworking/VimeoNetworking/Request.swift

```diff
@@ -164,31 +165,69 @@ public struct Request<ModelType: MappableResponse>
          retryPolicy: RetryPolicy? = nil)
     {
         self.method = method
+        
+        let (path, query) = path.splitLinkString()
         self.path = path
-        self.additionalParameters = parameters ?? [:]
-        self.page = page
-        self.itemsPerPage = itemsPerPage
+        
+        if let query = query,
+            let queryParameters = query.parametersFromQueryString()
+        {
+            var page = page
```

Comment:
Maybe possibly, but it would be tough because swift doesn't have runtime introspection so you can't go from `"page"` to `self.page` in code.  

These are good thoughts to funnel into a picture of what a swift JSON model object parser might look like though.  Swift 4 is already talking about better introspective features, and I for one can't wait 😁